R/ntfy.R

```diff
@@ -18,6 +18,39 @@ ntfy_server <- function(var = "NTFY_SERVER") {
   Sys.getenv(var)
 }
 
+#' Create HTTP headers for basic authentication
+#' 
+#' @param var_username environment variable in which the ntfy username is stored
+#' @param var_password environment variable in which the ntfy password is stored
+#' 
+#' @return an object with class `request` that is passed to [httr::POST()]
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' Sys.setenv("NTFY_USERNAME" = "example")
+#' Sys.setenv("NTFY_PASSWORD" = "super-secret-password")
+#' 
+#' # This generates the correct heading:
+#' ntfy_authorization()
+#' #> <request>
+#' #> Headers:
+#' #>   * Authorization: Basic ZXhhbXBsZTpzdXBlci1zZWNyZXQtcGFzc3dvcmQ
+#' }
+#' 
+ntfy_authorization <- function(var_username = "NTFY_USERNAME", var_password = "NTFY_PASSWORD") {
+  username = Sys.getenv(var_username)
+  password = Sys.getenv(var_password)
+  
+  if (username == "") {
+    httr::add_headers()
```

Comment:
What about adding an alias function 
```r
no_auth = httr::add_headers
```
that can alternatively be passed to `ntfy_send()` in place of `ntfy_authorization(var_username = "")` in order to disable auth (even though the env var is set)?

Another option would be to allow `ntfy_send(auth=FALSE)` and handle the logic there.

Reply:
Here I made `auth = NULL` by default, and no authorization attempt is made unless the user specifies `auth = ntfy_authorization()`: https://github.com/jonocarroll/ntfy/pull/12/commits/44d6c8db362db66ab0b6b3dd2f5a6bde41f95777